src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -55,7 +55,7 @@ public class RemoteUriWebService : IRemoteUriWebService {
             var remoteUri = GetRemoteUri(context.Request.Url, remoteBaseUrl);
             request.Headers.Add(CustomHttpHeaders.RTVSRequestedURL, remoteUri.ToString());
 
-            if (context.Request.InputStream.CanSeek && context.Request.InputStream.Length > 0) {
+            if (context.Request.ContentLength64 > 0) {
```

Comment:
Seek was needed to query the length from a stream for POST messages. Using the stream directly to determine the length requires additional conditions. For, example, when a file is uploaded, the stream is *not* seek-able but read-only. For post messages, the stream may be seek-able and length is only available when it is seek-able. `ContentLength64` stores the message body length for all request objects, and it is simpler. The issue here was caused by read-only case I mentioned above.